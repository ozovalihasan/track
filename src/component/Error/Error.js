import PropTypes from 'prop-types';
import styled from 'styled-components';

const Error = ({
  error,
}) => (
  <ErrorMain>
    {error}
    .
    {' '}
    Please try again.
  </ErrorMain>
);

const ErrorMain = styled.div`
font-family: 'Courier New', Courier, monospace;
text-align: center;
width: 100%;
position: absolute;
`;

Error.propTypes = {
  error: PropTypes.string.isRequired,
};

export default Error;
