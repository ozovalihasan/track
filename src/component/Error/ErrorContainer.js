import React from 'react';
import { useSelector } from 'react-redux';
import Error from './Error';

const ErrorContainer = () => {
  const error = useSelector(state => state.api.error);

  return (
    error
      ? (<Error error={error.message} />)
      : (<></>)
  );
};

export default ErrorContainer;
