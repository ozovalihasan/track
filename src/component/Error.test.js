import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Error from './Error';

const error = 'There is an error';

describe('<Error />', () => {
  it('links to root page', () => {
    render(
      <Error error={error} />,
    );
    expect(screen.getByText(/There is an error/i)).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Error error={error} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
