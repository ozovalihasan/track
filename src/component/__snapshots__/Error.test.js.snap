// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
<div
  className="sc-bdfBwQ Engni"
>
  There is an error
  .
   
  Please try again.
</div>
`;
